wgpu-hal/src/dynamic/device.rs

```diff
@@ -1,9 +1,14 @@
+// Box casts are needed, alternative would be a temporaries which are more verbose and not more expressive.
+#![allow(trivial_casts)]
+
```

Comment:
suggestion: This is actually not necessary; the offending lines of code can be fixed thusly:

```diff
-            .map(|b| Box::new(b) as Box<dyn DynCommandEncoder>)
+            .map(|b| -> Box<dyn DynCommandEncoder> { Box::new(b) })
```

Will push a fix myself anon.